process-engine/bpmn-studio
# WIP✨Safe New Diagrams Temporary

**Changes:**

1. Add functionality to create diagrams within a `Single Diagram` solution.
2. Set the `Single Diagram` solution as the default for new diagrams.
3. Display the count of unsaved diagrams and mark them on SolutionExplorer.
4. Don't persist unsaved diagrams on LocalStorage.
5. Add the possibility to save unsaved diagrams to a custom path.

**Issues:**

Closes #1401 
Closes #1397 

PR: #1394 

<img width="289" alt="Screenshot 2019-03-11 at 14 17 58" src="https://user-images.githubusercontent.com/17065920/54126718-80a16a00-4408-11e9-8c15-8d19f874abc0.png">

## How can others test the changes?

- checkout the branch
- start the studio and create a new diagram e.g. with `cmd+n`
- navigate to some other diagrams after that
- try to quit the studio (`cmd+q` or `cmd+w`)
- see that modal and try all of its buttons
- check that saving on design view works as expected, except with an unsaved diagram where you must set a path


## PR-Checklist

Please check the boxes in this list after submitting your PR:
